frontend/src/views/DashboardHeader.vue

```diff
@@ -301,68 +277,99 @@ export default {
       }),
       defineAction({
         id: "bb.navigation.databases",
         name: "Databases",
-        keywords: "db",
         shortcut: ["g", "d"],
-        section: "Navigation",
+        section: t("kbar.navigation"),
+        keywords: "navigation db",
         perform: () => router.push({ name: "workspace.database" }),
       }),
       defineAction({
         id: "bb.navigation.instances",
         name: "Instances",
         shortcut: ["g", "i"],
-        section: "Navigation",
+        section: t("kbar.navigation"),
+        keywords: "navigation",
         perform: () => router.push({ name: "workspace.instance" }),
       }),
       defineAction({
         id: "bb.navigation.environments",
         name: "Environments",
         shortcut: ["g", "e"],
-        section: "Navigation",
+        section: t("kbar.navigation"),
+        keywords: "navigation",
         perform: () => router.push({ name: "workspace.environment" }),
       }),
       defineAction({
         id: "bb.navigation.settings",
         name: "Settings",
         shortcut: ["g", "s"],
-        section: "Navigation",
+        section: t("kbar.navigation"),
+        keywords: "navigation",
         perform: () => router.push({ name: "setting.workspace.general" }),
       }),
       defineAction({
         id: "bb.navigation.inbox",
         name: "Inbox",
         shortcut: ["g", "m"],
-        section: "Navigation",
+        section: t("kbar.navigation"),
+        keywords: "navigation",
         perform: () => router.push({ name: "setting.inbox" }),
       }),
-    ];
+    ]);
     useRegisterActions(kbarActions);
 
-    const { availableLocales, locale } = useI18n();
     const storage = useLocalStorage("bytebase_options", {}) as any;
 
-    const toggleLocales = () => {
-      // TODO change to some real logic
-      const locales = availableLocales;
-      locale.value =
-        locales[(locales.indexOf(locale.value) + 1) % locales.length];
+    const setLocale = (lang: string) => {
+      locale.value = lang;
       storage.value = {
         appearance: {
-          language: locale.value,
+          language: lang,
         },
       };
     };
 
+    const toggleLocales = () => {
+      // TODO change to some real logic
+      const locales = availableLocales;
+      const nextLocale =
+        locales[(locales.indexOf(locale.value) + 1) % locales.length];
+      setLocale(nextLocale);
+    };
+
+    const i18nChangeAction = computed(() =>
+      defineAction({
+        id: "bb.preferences.locale",
+        section: t("kbar.preferences.common"),
+        name: t("kbar.preferences.change-language"),
+        keywords: "language lang locale",
+      })
+    );
+    const i18nActions = computed(() => [
+      i18nChangeAction.value,
+      ...availableLocales.map((lang) => {
+        return defineAction({
+          id: `bb.preferences.locale.${lang}`,
+          name: lang,
+          parent: "bb.preferences.locale",
```

Comment:
Didn't check our kbar library yet, but I pretty hope we can auto-generate all these parent-child relationships by id pattern in a further version, maybe by parsing the id path through `.`

And how about extract the `"bb.preferences.locale"` here? Seems repeated several times

Reply:
`defineAction` is now just a transparent function only for type inferring. But it is a very good suggestion to upgrade it as 
`defineAction<T>(params: Partial<Action<T>>, parent?: Action<any>) => Action<T>`
and
`defineActions<T>(params: Partial<Action<T>>[], parent?: Action<any>) => Action<T>[]`
in the future.

In next version I may refactor all "bb.*" actions to be created by a utility function like
```
createKbarAction<T>(
  namespace: KBarNamespace,
  params: Partial<Action<T>>,
  parent?: Action<any>
) => Action<T>
```
in which `KBarNamespace` is a constant enum, to avoid using string literals when defining actions.

Reply:
> And how about extract the "bb.preferences.locale" here? Seems repeated several times

done in latest commit. PTAL